xbmc/video/dialogs/GUIDialogVideoManagerVersions.cpp

```diff
@@ -464,22 +423,17 @@ bool CGUIDialogVideoManagerVersions::ConvertVideoVersion(const std::shared_ptr<C
 
   const int dbId{item->GetVideoInfoTag()->m_iDbId};
 
+  // Self-exclusion and decorate the items
   for (int i = 0; i < list.Size(); ++i)
   {
+    list[i]->SetLabel2(list[i]->GetVideoInfoTag()->m_strFileNameAndPath);
+
     if (list[i]->GetVideoInfoTag()->m_iDbId == dbId)
-    {
       list.Remove(i);
```

Comment:
Remove items in same loop causes not all items processed as `i < list.Size()` condition changes dynamically: this is the cause full paths are not (always) displayed. Depends on if item to remove is before or after to item to keep.

I propose this or similar:

```c++
  for (int i = 0; i < list.Size(); ++i)
    list[i]->SetLabel2(list[i]->GetVideoInfoTag()->m_strFileNameAndPath);

  const int videoAssetDbId = m_videoAsset->GetVideoInfoTag()->m_iDbId;
  for (int i = 0; i < list.Size(); ++i)
  {
    if (list[i]->GetVideoInfoTag()->m_iDbId == videoAssetDbId)
    {
      list.Remove(i);
      break;
    }
  }
```

In this way full path is displayed always:

![Captura de pantalla 2023-12-30 102714](https://github.com/xbmc/xbmc/assets/58434170/871bbe06-644d-4a2c-9e44-9aa093c28115)

Reply:
Thanks, I didn't notice the side effect when I combined the loops.
Kind of ugly but works?
```
  for (int i = 0; i < list.Size(); ++i)
  {
    list[i]->SetLabel2(list[i]->GetVideoInfoTag()->m_strFileNameAndPath);

    if (list[i]->GetVideoInfoTag()->m_iDbId == videoAssetDbId)
    {
      list.Remove(i);
      --i;
    }
  }
```

Reply:
I would go with:

```c++
  for (int i = 0; i < list.Size();)
  {
    list[i]->SetLabel2(list[i]->GetVideoInfoTag()->m_strFileNameAndPath);

    if (list[i]->GetVideoInfoTag()->m_iDbId == videoAssetDbId)
      list.Remove(i);
    else
      ++i;
  }
```

Reply:
> Kind of ugly but works?

Yes, this code is hard to understand. The basic concept is that when item is removed same `i` position is occupied with other item that needs to be processed (add path). Then first remove, second process same `i` position in same iteration:

Not tested but this "should" work:
 
```c++
  for (int i = 0; i < list.Size(); ++i)
  {
    if (list[i]->GetVideoInfoTag()->m_iDbId == videoAssetDbId)
      list.Remove(i);

    if (i < list.Size())
      list[i]->SetLabel2(list[i]->GetVideoInfoTag()->m_strFileNameAndPath);
  }
```

Reply:
Looks good to me. The single loop will make a difference for the Browse library with thousands of items.

Reply:
What I suggested is well proved and done like tat in many other projects. IMO easy to read and understand.

Reply:
The trick is to medial increase the counter.

Reply:
yeah it's a disguised while loop. I hesitated to rewrite it as such.

@thexai that suggestion doesn't work because it will still skip an item. That's why I added the --i in my ugly version.